src/suggesters/command-suggester.ts

```diff
@@ -0,0 +1,38 @@
+import {TextInputSuggest} from './suggest';
+import type {App, Command} from 'obsidian';
+
+export default class CommandSuggester extends TextInputSuggest<Command> {
+  constructor(
+    public app: App,
+    public inputEl: HTMLInputElement,
+  ) {
+    super(app, inputEl);
+  }
+
+  getSuggestions(input_str: string): Command[] {
+    // @ts-ignore
+    const all_commands = this.app.commands.listCommands();
```

Comment:
Why do we need `@ts-ignore` here? It looks like we could just add the missing type info to the module declaration in  `main.ts`: https://github.com/platers/obsidian-linter/blob/master/src/main.ts#L32-L40

So something like the following (my syntax for method declarations in an interface may be wrong):
``` Typescript
declare module 'obsidian' {
  // eslint-disable-next-line no-unused-vars
  interface App {
    commands: {
        executeCommandById(id: string): void;
        commands: {
            'editor:save-file': {
                callback(): void;
            };
        listCommands: function(): Command[],
        };
    };
  }
}
```

Reply:
You are right!

Reply:
But I have doubts here, directly import type from obsidian.d.ts, maybe need not to declare an obsidian package?

Reply:
Some properties on the actual interface are not exported. This seems to be one of them. Thus we have to add a local module declaration to add in the missing properties. It is generally only meant for use in cases where you would otherwise need to do a type cast or `@ts-ignore` to get eslint to know it exits.